File: app/models/qdm/element_definition.rb
module QDM
  # app/models/qdm/element_definition.rb
  class ElementDefinition < DataElement
    include Mongoid::Document
    embedded_in :patient
    field :path
    field :representation
    field :sliceName
    field :sliceIsConstraining
    field :label
    field :code
    field :slicing
    field :short
    field :definition
    field :comment
    field :requirements
    field :alias
    field :min
    field :max
    field :base
    field :contentReference
    field :type
    field :defaultValue
    field :meaningWhenMissing
    field :orderMeaning
    field :fixed
    field :pattern
    field :example
    field :minValue
    field :maxValue
    field :maxLength
    field :condition
    field :constraint
    field :mustSupport
    field :isModifier
    field :isModifierReason
    field :isSummary
    field :binding
    field :mapping
    field :qdmVersion, type: String, default: '4.0.0'
  end
end

File: app/assets/javascripts/ElementDefinition.js
const mongoose = require('mongoose');
const { DataElementSchema } = require('./basetypes/DataElement');
const Code = require('./basetypes/Code');
const Interval = require('./basetypes/Interval');
const Quantity = require('./basetypes/Quantity');
const DateTime = require('./basetypes/DateTime');
const { ComponentSchema } = require('./Component');
const { FacilityLocationSchema } = require('./FacilityLocation');
const { IdSchema } = require('./Id');
const Any = require('./basetypes/Any');

const [Number, String] = [
  mongoose.Schema.Types.Number,
  mongoose.Schema.Types.String,
];

const ElementDefinitionSchema = DataElementSchema({
  path: Any,
  representation: Any,
  sliceName: Any,
  sliceIsConstraining: Any,
  label: Any,
  code: Any,
  slicing: Any,
  short: Any,
  definition: Any,
  comment: Any,
  requirements: Any,
  alias: Any,
  min: Any,
  max: Any,
  base: Any,
  contentReference: Any,
  type: Any,
  defaultValue: Any,
  meaningWhenMissing: Any,
  orderMeaning: Any,
  fixed: Any,
  pattern: Any,
  example: Any,
  minValue: Any,
  maxValue: Any,
  maxLength: Any,
  condition: Any,
  constraint: Any,
  mustSupport: Any,
  isModifier: Any,
  isModifierReason: Any,
  isSummary: Any,
  binding: Any,
  mapping: Any,
  qdmVersion: { type: String, default: '4.0.0' },
  _type: { type: String, default: 'ElementDefinition' },

});

module.exports.ElementDefinitionSchema = ElementDefinitionSchema;
module.exports.ElementDefinition = mongoose.model('ElementDefinition', ElementDefinitionSchema);
